From 7ab66eb3a8c35a1bef2f0b85bde231c91521d04b Mon Sep 17 00:00:00 2001
From: "Todd C. Miller" <Todd.Miller@sudo.ws>
Date: Sun, 19 Sep 2021 13:58:56 -0600
Subject: [PATCH] sudo -i: missing NULL terminator when moving argv to make
 room for --login Fixes a potential crash for "sudo -i" when the target user
 has bash as the shell (which needs the --login option).  Bug #998.

---
 plugins/sudoers/sudoers.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/plugins/sudoers/sudoers.c b/plugins/sudoers/sudoers.c
index 4fa323975..51376f970 100644
--- a/plugins/sudoers/sudoers.c
+++ b/plugins/sudoers/sudoers.c
@@ -725,7 +725,7 @@ sudoers_policy_main(int argc, char * const argv[], int pwflag, char *env_add[],
 	if (NewArgc > 1 && strcmp(NewArgv[0], "-bash") == 0 &&
 	    strcmp(NewArgv[1], "-c") == 0) {
 	    /* We allocated extra space for the --login above. */
-	    memmove(&NewArgv[2], &NewArgv[1], sizeof(char *) * (NewArgc - 1));
+	    memmove(&NewArgv[2], &NewArgv[1], sizeof(char *) * NewArgc);
 	    NewArgv[1] = "--login";
 	    NewArgc++;
 	}
